libnmstate/nm/applier.py

```diff
@@ -304,23 +338,39 @@ def prepare_proxy_ifaces_desired_state(ifaces_desired_state):
         new_ifaces_desired_state.append(port_iface_desired_state)
         # The "visible" slave/interface needs to point to the port profile
         iface_desired_state[MASTER_METADATA] = port_iface_desired_state[
-            Interface.NAME
+            IFACE_NAME_METADATA
         ]
         iface_desired_state[MASTER_TYPE_METADATA] = ovs.PORT_TYPE
     return new_ifaces_desired_state
 
 
 def _create_ovs_port_iface_desired_state(iface_desired_state, port_options):
+    port_name = ovs.PORT_PROFILE_PREFIX + iface_desired_state[Interface.NAME]
     return {
-        "name": ovs.PORT_PROFILE_PREFIX + iface_desired_state[Interface.NAME],
-        "type": ovs.PORT_TYPE,
-        "state": iface_desired_state[Interface.STATE],
+        Interface.NAME: port_name,
+        Interface.TYPE: ovs.PORT_TYPE,
+        Interface.STATE: iface_desired_state[Interface.STATE],
+        OvsB.OPTIONS_SUBTREE: port_options,
         MASTER_METADATA: iface_desired_state[MASTER_METADATA],
         MASTER_TYPE_METADATA: iface_desired_state[MASTER_TYPE_METADATA],
-        "options": port_options,
+        IFACE_NAME_METADATA: _generate_hash_iface_name(port_name),
     }
 
 
+def _generate_hash_iface_name(name):
+    """
+    Given a name, generate a hash string that may be used as an interface name.
+
+    The OVS port does not have an actual interface in the kernel, but
+    its length is still limited by NM connection.interface-name.
+    https://bugzilla.redhat.com/1788432
+    As a workaround, hash the full name and use it as an alternative device
+    name (assigned as a metadata). The profile/connection name format has
+    an OVS PORT prefix
+    """
+    return f"{hashlib.md5(name.encode()).hexdigest():.15}"
```

Comment:
Could you please use sha256 here? Even though using md5 is not a problem here it requires people to evaluate if it is ok or not. Using sha256 should not cause any trouble. Also you could use base64 instead of hex to reduce the probability of a collisions slightly more.

Reply:
sha256 for a name hashing? That sounds an overkill. Anyway, I do not mind, it cannot do anything wrong except questioning why the coder exaggerated.

I am not sure I understand what you suggests about `base64`.

Reply:
Currently the hash value is encoded in hexadecimal, therefore 15 characters mean its 15 * 4 = 60 bits, with base64 it would be 15 * 6 = 90 bits.
Also I just noticed that using a format string seems to be overkill, you could just slice it with '[:15]` and the 15 should probably be a constant like `MAX_INTERFACE_NAME_LENGTH` to clarify why it is 15.

Reply:
You have 4 comments on a single line and I am not sure if I understand them all.
It can help if you present (through code) the alternative and avoid misunderstandings.

For example, I do not understand how you represent a char of 6 bit using an alphanumeric representation. A hex seems the simplest and easiest.

Reply:
Proposal:

```
MAXIMUM_INTERFACE_LENGTH=15
return base64.b64encode(hashlib.sha256(name.encode()).digest(), altchars=b"-_")[:MAXIMUM_INTERFACE_LENGTH]
```

Reply:
Done.